### PR TITLE
Migrate Python tests to not depend on fix sharding for file output

### PR DESCRIPTION
**Pull request by charlesccychen**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @aaltay

File: sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py [hunk 1]
```diff
@@ -39,7 +40,7 @@ def create_temp_file(self, contents):
 
   def get_wordcount_results(self, temp_path):
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py]: Could you rename `temp_path` to `result_path`?

File: sdks/python/apache_beam/testing/util.py [hunk 2]
```diff
@@ -32,6 +36,8 @@
     'assert_that',
     'equal_to',
     'is_empty',
+    # open_shards is internal and has no backwards compatibility guarantees.
```
Comment (aaltay) [inline sdks/python/apache_beam/testing/util.py]: You can also use `@experimental` wrapper.

File: sdks/python/apache_beam/testing/util.py [hunk 3]
```diff
@@ -105,3 +111,12 @@ def default_label(self):
...
   actual | AssertThat()  # pylint: disable=expression-not-assigned
+
+
+def open_shards(glob_pattern):
+  """Returns a composite file of all shards matching the given glob pattern."""
+  with tempfile.NamedTemporaryFile(delete=False) as f:
+    for shard in glob.glob(glob_pattern):
+      f.write(file(shard).read())
+    concatenated_file_name = f.name
+  return file(concatenated_file_name, 'rb')
```
Comment (aaltay) [inline sdks/python/apache_beam/testing/util.py]: Is it possible to not leave these temp files after each run?

File: sdks/python/apache_beam/examples/cookbook/mergecontacts_test.py [hunk 4]
```diff
@@ -92,6 +93,8 @@ def normalize_tsv_results(self, tsv_data):
     lines_in = tsv_data.strip().split('\n')
     lines_out = []
     for line in lines_in:
+      if not line:
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/cookbook/mergecontacts_test.py]: Why is this change needed?

Comment (charlesccychen) [inline sdks/python/apache_beam/examples/cookbook/mergecontacts_test.py]: Thanks, fixed.

File: sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py (hunk 1 above)
Comment (charlesccychen) [inline sdks/python/apache_beam/examples/cookbook/multiple_output_pardo_test.py]: Yes, done.

File: sdks/python/apache_beam/testing/util.py (hunk 2 above)
Comment (charlesccychen) [inline sdks/python/apache_beam/testing/util.py]: Thanks, I added this as well.

File: sdks/python/apache_beam/testing/util.py (hunk 3 above)
Comment (charlesccychen) [inline sdks/python/apache_beam/testing/util.py]: This is an issue for all tests at the moment.  I have filed https://issues.apache.org/jira/browse/BEAM-2413.

**Comment by aaltay** [comment]
LGTM
